appscale/tools/node_layout.py

```diff
@@ -737,7 +740,20 @@ def from_locations_json_list(self, locations_nodes_list):
             # Locations JSON is incorrect if we get here.
             return None
           break
-
+    if len(nodes_copy) > 0:
```

Comment:
To reduce nesting, you can remove this check. If you pop from `nodes_copy`, you can catch `IndexError`.